### PR TITLE
Move md5 hash to message instead of all messages

### PR DESCRIPTION
**Pull request by kfiroo**

### CODE REVIEW

**Comment by kfiroo** [comment]
@FaridSafi what do you think?

File: src/MessageContainer.js [hunk 1]
```diff
@@ -1,29 +1,50 @@
-import React, { Component, PropTypes } from 'react';
+import React, {Component, PropTypes} from 'react';
+
+import ReactNative, {
+  View,
+  ListView,
+  UIManager
```
Comment (FaridSafi) [inline src/MessageContainer.js]: is UIManager used ?

File: src/MessageContainer.js [hunk 2]
```diff
@@ -47,46 +68,64 @@ export default class MessageContainer extends Component {
...
+      </View>
     );
   }
 }
 
 MessageContainer.defaultProps = {
   messages: [],
   user: {},
-  footer: null,
+  renderFooter: null,
```
Comment (FaridSafi) [inline src/MessageContainer.js]: I guess it should also be a default props of GiftedChat.js

File: src/MessageContainer.js (hunk 1 above)
Comment (kfiroo) [inline src/MessageContainer.js]: Nope, I tried to make the content stick to the top when its smaller then the container, it didn't work that well so I removed it.
I will remove the left over reference

**Comment by FaridSafi** [comment]
I will do the following tests with a device :

While having 400 messages displayed
- Typing a character should feels reactive
- Typing a character in the text input should not trigger ListView rowHasChanged call

File: src/MessageContainer.js (hunk 2 above)
Comment (kfiroo) [inline src/MessageContainer.js]: fixed

**Comment by FaridSafi** [comment]
Typing a character in the text input should not trigger ListView rowHasChanged call
=> rowHasChanged is called for each character typed

**Comment by kfiroo** [comment]
Ok, will check

**Comment by kfiroo** [comment]
@FaridSafi does it return `true` for each row?

**Comment by FaridSafi** [comment]
it returns `false`

**Comment by kfiroo** [comment]
@FaridSafi Then we are good :)
The list view will attempt to render the rows only if `rowHasChanged` returns `true`

**Comment by FaridSafi** [comment]
yes but it slows the typing

**Comment by kfiroo** [comment]
I will try to fix it

**Comment by FaridSafi** [comment]
What do you think about keeping the hash at the messages list level and also at the message level ?

**Comment by FaridSafi** [comment]
`initialListSize={this.props.messages.length}` is needed to display more than 22 rows
I guess `pageSize` will be needed also, it should be the amount of displayable rows in the screen (without scrolling)

I have to say that I am not convinced yet by the use of ListView

**Comment by FaridSafi** [comment]
rowHasChanged calls for each typed character can create this warning

simulator screen shot 01 aug 2016 16 47 53

**Comment by kfiroo** [comment]
@FaridSafi ListView is supposed to handle large sets of data much better then static rendering, yet for now I agree this is not stable enough to merge.
I will play around with it and let you know if I got anything interesting :)

**Comment by kfiroo** [comment]
@FaridSafi closing this and submitting a new one without the unneeded renders

**Pull request closed without merging**